source/aggregation/stages.txt

```diff
@@ -0,0 +1,357 @@
+.. _csharp-aggregation-stages:
+
+===========================
+Aggregation Pipeline Stages
+===========================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: dotnet, code example, transform, pipeline
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+On this page, you can learn how to create an aggregation pipeline and pipeline stages  
+by using methods in the {+driver-short+}.
+
+Build an Aggregation Pipeline
+-----------------------------
+
+You can use the {+driver-short+} to build an aggregation pipeline by using builder
+methods or BSON documents. See the following sections to learn more about each of these
+approaches.
+
+.. _csharp-aggregation-stages-builder:
+
+Builder Methods
+~~~~~~~~~~~~~~~
+
+You can build a type-safe aggregation pipeline in the following ways:
+
+- Construct an ``EmptyPipelineDefinition<TDocument>`` object. Chain calls from this object
+  to the relevant aggregation methods. Then, pass the pipeline object to the
+  ``IMongoCollection<TDocument>.Aggregate()`` method.
+
+- Call the ``IMongoCollection<TDocument>.Aggregate()`` method. Chain calls from this
+  method call to the relevant aggregation methods.
+
+Select the :guilabel:`EmptyPipelineDefinition`
+or :guilabel:`Aggregate` tab to see the corresponding code for each approach:
+
+.. tabs::
+
+   .. tab:: EmptyPipelineDefinition
+      :tabid: empty-pipeline-definition
+
+      .. code-block:: csharp
+
+         // Defines the aggregation pipeline
+         var pipeline = new EmptyPipelineDefinition<Movie>()
+            .Match(...)
+            .Group(...)
+            .Merge(...);
+
+         // Executes the aggregation pipeline
+         var results = collection.Aggregate(pipeline);
+
+   .. tab:: Aggregate
+      :tabid: aggregate
+
+      .. code-block:: csharp
+
+         // Defines and executes the aggregation pipeline
+         var results = collection.Aggregate()
+            .Match(...)
+            .Group(...)
+            .Merge(...);
+
+.. _csharp-aggregation-stages-bsondocument:
+
+BsonDocument
+~~~~~~~~~~~~
+
+Some aggregation stages don't have corresponding methods in the {+driver-short+}.
+To add these stages to your pipeline, use ``BsonDocument`` objects or string literals
+to construct a stage in the Query API syntax. Then, pass the BSON document to the
+``PipelineDefinitionBuilder.AppendStage()`` method. This syntax supports all stages
+in the aggregation pipeline, but doesn't provide type hints or type safety.
+
+The following code example shows how to add ``$unset``, an aggregation
+stage without a corresponding builder method, to an empty aggregation pipeline:
+
+.. code-block:: csharp
+
+   var pipeline = new EmptyPipelineDefinition<BsonDocument>()
+           .AppendStage<BsonDocument, BsonDocument, BsonDocument>("{ $unset: "field1" }");
```

Comment:
S: text highlighting weird, do you need to escape the characters as there's multiple sets of " marks?